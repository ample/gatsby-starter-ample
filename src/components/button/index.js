import React from "react"

import Component from "./component"
import fixtures from "./fixtures"
import transform from "./transformer"

const Button = props => <Component {...transform(props)} />

export default Button

export { Component as component, fixtures, transform as transformer }
