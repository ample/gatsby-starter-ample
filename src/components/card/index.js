import React from "react"
import PropTypes from "prop-types"
import classNames from "classnames"

import Link from "~components/link"
import Button from "~components/button"
import Image from "~components/image"

import styles from "./styles.module.scss"

const Card = ({ body, button, image, theme, url }) => {
  const classes = classNames(styles.card, {
    [styles[theme]]: theme
  })

  let card = (
    <div className={classes}>
      {image && <Image src={image} />}
      {body && <div dangerouslySetInnerHTML={{ __html: body }} />}
      {button && <Button to={button.url}>{button.label}</Button>}
    </div>
  )

  if (url) {
    card = <Link to={url}>{card}</Link>
  }

  return card
}

Card.propTypes = {
  /**
   * Specifies the content
   * that is displayed under the heading
   */
  body: PropTypes.string,
  /**
   * Specifies the label and url
   * if the card has a button
   */
  button: PropTypes.exact({
    label: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
  }),
  /**
   * Specifies the image src and alt text
   */
  image: PropTypes.oneOfType([PropTypes.string, PropTypes.object]).isRequired,
  /**
   * Specifies the card theme
   */
  theme: PropTypes.oneOf(["theme_1", "theme_2"]),
  /**
   * Specifies the url for the card
   *
   * NOTE: Only use this prop,
   * if the card DOES NOT have a button
   * This will make the entire card a link
   */
  url: PropTypes.string
}

Card.defaultProps = {}

export default Card
