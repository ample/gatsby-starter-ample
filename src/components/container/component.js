import React from "react"
import PropTypes from "prop-types"
import classNames from "classnames"

import styles from "./styles.module.scss"

import Block from "../block"

const Container = ({ className, children, blocks, config }) => {
  const wrapperClasses = classNames(styles.container, {
    [className]: className,
    [`mb-${config.margin_bottom}`]: config.margin_bottom
  })

  return (
    <div className={wrapperClasses}>
      <div className={styles.content}>
        {children}

        {blocks && blocks.map((column, idx) => <Block key={idx} {...column} />)}
      </div>
    </div>
  )
}

Container.propTypes = {
  /**
   * A CSS class adding to the wrapping element.
   */
  className: PropTypes.string,
  /**
   * For use in templates when more control is needed.
   */
  children: PropTypes.node,
  /**
   * An array of components that get passed on to the <Block />.
   */
  blocks: PropTypes.arrayOf(PropTypes.object),
  /**
   * An object that controls the styling of the outer container on screen.
   */
  config: PropTypes.shape({
    margin_bottom: PropTypes.string
  })
}

Container.defaultProps = {
  blocks: [],
  config: {}
}

export default Container
