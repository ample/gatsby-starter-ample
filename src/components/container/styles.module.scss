@use '../../styles/global' as *;

$column-gutter: 40px;

.container {
  margin-left: auto;
  margin-right: auto;
  max-width: 1000px + $column-gutter;

  .content {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: space-between;
    margin-left: -($column-gutter / 2);
    margin-right: -($column-gutter / 2);
    width: 100%;
  }
}
