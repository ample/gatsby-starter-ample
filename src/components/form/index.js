import React from "react"

import Component from "./component"
import fixtures from "./fixtures"
import transform from "./transformer"

const Form = props => <Component {...transform(props)} />

export default Form

export { Component as component, fixtures, transform as transformer }
