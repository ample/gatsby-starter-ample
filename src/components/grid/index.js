import React from "react"
import PropTypes from "prop-types"
import classNames from "classnames"

import styles from "./styles.module.scss"

const Grid = ({ alignment, children, layout }) => {
  const classes = classNames(styles.grid, {
    [styles[`alignment_${alignment}`]]: alignment,
    [styles[`layout_is_${layout}`]]: layout
  })

  return <div className={classes}>{children}</div>
}

Grid.propTypes = {
  /**
   * Specifies how each grid item should be aligned within the grid
   */
  alignment: PropTypes.oneOf(["top", "bottom", "center"]),
  /**
   * Specifies anything between the opening and closing tag.
   */
  children: PropTypes.node,
  /**
   * Specifies the width of each grid item within the grid on larger screens
   */
  layout: PropTypes.oneOf(["1/4", "1/3", "1/2"])
}

Grid.defaultProps = {
  layout: "1/2"
}

export default Grid
