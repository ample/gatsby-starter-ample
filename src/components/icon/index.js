import React, { Fragment } from "react"
import PropTypes from "prop-types"
import icons from "./icons"

const Icon = ({ name }) => {
  return <Fragment>{icons[name]}</Fragment>
}

Icon.propTypes = {
  /**
   * Specifies the name of the icon
   */
  name: PropTypes.string.isRequired
}

Icon.defaultProps = {}

export default Icon
