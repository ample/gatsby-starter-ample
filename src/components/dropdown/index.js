import component from "./component"
import fixtures from "./fixtures"

import DropdownMenu from "./menu"
import DropdownTrigger from "./trigger"

const Dropdown = component

export default component

export { component, fixtures, Dropdown, DropdownMenu, DropdownTrigger }
