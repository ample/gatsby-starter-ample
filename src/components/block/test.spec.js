import React from "react"
import renderer from "react-test-renderer"

import Block from "."
import { fixtures } from "@src/components/container"

describe("Block", () => {
  it("returns null when no data", () => {
    const tree = renderer.create(<Block />).toJSON()
    expect(tree).toEqual(null)
  })
  it("returns null when no template property", () => {
    const tree = renderer.create(<Block template={null} />).toJSON()
    expect(tree).toEqual(null)
  })
  it("returns null when no component mapping exists", () => {
    const tree = renderer.create(<Block template={"WRONG-COMP-NAME"} />).toJSON()
    expect(tree).toEqual(null)
  })
  it("renders properly with valid props", () => {
    const tree = renderer.create(<Block {...fixtures.default.blocks[0]} />).toJSON()
    expect(tree).not.toEqual(null)
    expect(tree).toMatchSnapshot()
  })
})
