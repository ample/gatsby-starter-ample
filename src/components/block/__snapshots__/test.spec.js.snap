// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block renders properly with valid props 1`] = `
<a
  className="button"
  href="/"
>
  Hello World
</a>
`;
